### ISSUE TITLE
Crash at startup in ue5.0.3

### ISSUE THREAD

**Issue by QuentinGarou** [millicast/millicast-player-unreal-engine-plugin#18]
I've rebuilt your plugin successfully but I'm having a crash on startup because of SlateRegistry.
Here is the Crash report:
```
Assertion failed: !FindSlateStyle(SlateStyleName) [File:D:\build\++UE5\Sync\Engine\Source\Runtime\SlateCore\Private\Styling\SlateStyleRegistry.cpp] [Line: 21]

UnrealEditor_SlateCore!FSlateStyleRegistry::RegisterSlateStyle() [D:\build\++UE5\Sync\Engine\Source\Runtime\SlateCore\Private\Styling\SlateStyleRegistry.cpp:21]
UnrealEditor_MillicastPlayerEditor!FMillicastPlayerEditorModule::RegisterStyle() [C:\Users\dev\source\repos\GarouCore\Plugins\Milicast-player\Source\MillicastPlayerEditor\Private\MillicastPlayerEditorModule.cpp:77]
UnrealEditor_Core!FModuleManager::LoadModuleWithFailureReason() [D:\build\++UE5\Sync\Engine\Source\Runtime\Core\Private\Modules\ModuleManager.cpp:579]
UnrealEditor_Projects!FModuleDescriptor::LoadModulesForPhase() [D:\build\++UE5\Sync\Engine\Source\Runtime\Projects\Private\ModuleDescriptor.cpp:690]
UnrealEditor_Projects!FPluginManager::TryLoadModulesForPlugin() [D:\build\++UE5\Sync\Engine\Source\Runtime\Projects\Private\PluginManager.cpp:1583]
UnrealEditor_Projects!FPluginManager::LoadModulesForEnabledPlugins() [D:\build\++UE5\Sync\Engine\Source\Runtime\Projects\Private\PluginManager.cpp:1658]
UnrealEditor!FEngineLoop::Init() [D:\build\++UE5\Sync\Engine\Source\Runtime\Launch\Private\LaunchEngineLoop.cpp:4287]
UnrealEditor_UnrealEd!EditorInit() [D:\build\++UE5\Sync\Engine\Source\Editor\UnrealEd\Private\UnrealEdGlobals.cpp:111]
UnrealEditor!GuardedMain() [D:\build\++UE5\Sync\Engine\Source\Runtime\Launch\Private\Launch.cpp:159]
UnrealEditor!GuardedMainWrapper() [D:\build\++UE5\Sync\Engine\Source\Runtime\Launch\Private\Windows\LaunchWindows.cpp:147]
UnrealEditor!LaunchWindowsStartup() [D:\build\++UE5\Sync\Engine\Source\Runtime\Launch\Private\Windows\LaunchWindows.cpp:283]
UnrealEditor!WinMain() [D:\build\++UE5\Sync\Engine\Source\Runtime\Launch\Private\Windows\LaunchWindows.cpp:330]
UnrealEditor!__scrt_common_main_seh() [d:\a01\_work\6\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288]
kernel32
ntdll
```

**Comment by TroubledPrawn** [comment]
Hey, just wondering how you managed to rebuild the plugin successfully? I have not managed to do it

**Comment by QuentinGarou** [comment]
You need to create a c++ class in your project if it's a blueprinit only project.
Follow this: https://docs.unrealengine.com/5.0/en-US/managing-game-code-in-unreal-engine/

**Comment by QuentinGarou** [comment]
Quick note for the crash: opening the editor in Debug mode from the IDE just triggers a breakpoint and goes on to open it....

**Comment by TroubledPrawn** [comment]
Also crashing on running in the editor, but seems we have different issues. If I change the stream URL to something other than the suggested "https://director.millicast.com/api/director/subscribe" Then the engine does not crash, but obviously the stream does not work. 

[UE5crashreport.txt](https://github.com/millicast/millicast-player-unreal-engine-plugin/files/9310715/UE5crashreport.txt)

**Comment by lukehb** [comment]
Hi @QuentinGarou I wasn't able to repro your crash on Windows using UE 5.0.3 and building from the latest commit on the `main` branch - I can launch my project with the player plugin enabled. Can you let me know the following:

- Are you using Windows or Linux?
- Is this an UE launcher build or a source build of UE?
- Are you building from the `main` branch?
- Is there any extra steps to repro?
- Can you share your full log with me (should be under Saved/ )?

**Comment by lukehb** [comment]
@TroubledPrawn I agree, you seem to be hitting a different issue. Can I get you to open a new issue and answer the same questions as above and provide a minimal project or set of steps that repros the crash :)

**Comment by TroubledPrawn** [comment]
@lukehb I was about to post a new issue but I managed to get it working and stop the crashes. I think the guide for implementation in UE could be updated to be a bit more specific as the issue I was having was from adding the audio and video consumer inside the BP. 

When you create the consumer variables, you have to also create a new millicastTexture2DPlayer. I didnt notice this for quite a while, since I was under the impression that the 2d texture I had already created and applied to a plane was doing this. 

![image](https://user-images.githubusercontent.com/33198146/184918580-918bb719-d14d-4f1f-bdc7-02152c2a854f.png)

**Comment by QuentinGarou** [comment]
@lukehb I think the issue is really similar to this issue on the other plugin: [Issue](https://github.com/millicast/millicast-publisher-unreal-engine-plugin/issues/2)
I've noticed that it only happens when I run it from Rider though, not when I run it by double clicking on the uproject itself.

**Comment by QuentinGarou** [comment]
Also here are your answers: 
- I am using Windows 10
- It's on the launcher build of UE5.0.3
- I built this plugin from the main branch
- No extra step than just click run from Rider. A good repro step if you don't use it could running this command: 
%PathToUnrealEditor.exe% %PathToUProjectFile% -skipcompile
- Here are the latest lines of the log: 

> [2022.08.18-15.58.55:858][  0]LogCUDA: Display: Initialising CUDA API...
> [2022.08.18-15.58.55:872][  0]LogCUDA: Display: CUDA API initialised successfully.
> [2022.08.18-15.58.55:872][  0]LogCUDA: Display: Found 1 CUDA capable devices.
> [2022.08.18-15.58.55:872][  0]LogCUDA: Display: Found device 0 called "NVIDIA GeForce RTX 3070 Laptop GPU".
> [2022.08.18-15.58.55:872][  0]LogCUDA: Display: Attempting to create CUDA context on GPU Device 0...
> [2022.08.18-15.58.55:965][  0]LogCUDA: Display: Created CUDA context on device "NVIDIA GeForce RTX 3070 Laptop GPU".
> [2022.08.18-15.58.55:996][  0]LogEncoderNVENC: Created D3D11 device for NVENC on 'NVIDIA GeForce RTX 3070 Laptop GPU'.
> [2022.08.18-15.58.56:089][  0]LogLevelSequenceEditor: LevelSequenceEditor subsystem initialized.
> [2022.08.18-15.58.56:359][  0]LogOutputDevice: Warning: 
> 
> Script Stack (0 frames):
> 
> [2022.08.18-15.58.56:360][  0]LogWindows: Windows GetLastError: The operation completed successfully. (0)
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: === Critical error: ===
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: 
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: Assertion failed: !FindSlateStyle(SlateStyleName) [File:D:\build\++UE5\Sync\Engine\Source\Runtime\SlateCore\Private\Styling\SlateStyleRegistry.cpp] [Line: 21] 
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: 
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: 
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: 
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: [Callstack] 0x00007ffbb432caab UnrealEditor-SlateCore.dll!FSlateStyleRegistry::RegisterSlateStyle() [D:\build\++UE5\Sync\Engine\Source\Runtime\SlateCore\Private\Styling\SlateStyleRegistry.cpp:21]
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: [Callstack] 0x00007ffb319533c9 UnrealEditor-MillicastPlayerEditor.dll!FMillicastPlayerEditorModule::RegisterStyle() [C:\Users\dev\source\repos\GarouCore\Plugins\Core\Millicast\Milicast-player\Source\MillicastPlayerEditor\Private\MillicastPlayerEditorModule.cpp:77]
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: [Callstack] 0x00007ffbb5e8ea4d UnrealEditor-Core.dll!FModuleManager::LoadModuleWithFailureReason() [D:\build\++UE5\Sync\Engine\Source\Runtime\Core\Private\Modules\ModuleManager.cpp:579]
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: [Callstack] 0x00007ffbd998aaf6 UnrealEditor-Projects.dll!FModuleDescriptor::LoadModulesForPhase() [D:\build\++UE5\Sync\Engine\Source\Runtime\Projects\Private\ModuleDescriptor.cpp:690]
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: [Callstack] 0x00007ffbd999de4a UnrealEditor-Projects.dll!FPluginManager::TryLoadModulesForPlugin() [D:\build\++UE5\Sync\Engine\Source\Runtime\Projects\Private\PluginManager.cpp:1583]
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: [Callstack] 0x00007ffbd998a8b0 UnrealEditor-Projects.dll!FPluginManager::LoadModulesForEnabledPlugins() [D:\build\++UE5\Sync\Engine\Source\Runtime\Projects\Private\PluginManager.cpp:1658]
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: [Callstack] 0x00007ff62fbf2d04 UnrealEditor.exe!FEngineLoop::Init() [D:\build\++UE5\Sync\Engine\Source\Runtime\Launch\Private\LaunchEngineLoop.cpp:4287]
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: [Callstack] 0x00007ffb67f618b3 UnrealEditor-UnrealEd.dll!EditorInit() [D:\build\++UE5\Sync\Engine\Source\Editor\UnrealEd\Private\UnrealEdGlobals.cpp:111]
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: [Callstack] 0x00007ff62fc10cb1 UnrealEditor.exe!GuardedMain() [D:\build\++UE5\Sync\Engine\Source\Runtime\Launch\Private\Launch.cpp:159]
> [2022.08.18-15.58.58:148][  0]LogWindows: Error: [Callstack] 0x00007ff62fc10e8a UnrealEditor.exe!GuardedMainWrapper() [D:\build\++UE5\Sync\Engine\Source\Runtime\Launch\Private\Windows\LaunchWindows.cpp:147]
> [2022.08.18-15.58.58:149][  0]LogWindows: Error: [Callstack] 0x00007ff62fc13c4d UnrealEditor.exe!LaunchWindowsStartup() [D:\build\++UE5\Sync\Engine\Source\Runtime\Launch\Private\Windows\LaunchWindows.cpp:283]
> [2022.08.18-15.58.58:149][  0]LogWindows: Error: [Callstack] 0x00007ff62fc25564 UnrealEditor.exe!WinMain() [D:\build\++UE5\Sync\Engine\Source\Runtime\Launch\Private\Windows\LaunchWindows.cpp:330]
> [2022.08.18-15.58.58:149][  0]LogWindows: Error: [Callstack] 0x00007ff62fc28736 UnrealEditor.exe!__scrt_common_main_seh() [d:\a01\_work\6\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288]
> [2022.08.18-15.58.58:149][  0]LogWindows: Error: [Callstack] 0x00007ffc04097034 KERNEL32.DLL!UnknownFunction []
> [2022.08.18-15.58.58:149][  0]LogWindows: Error: [Callstack] 0x00007ffc05882651 ntdll.dll!UnknownFunction []
> [2022.08.18-15.58.58:149][  0]LogWindows: Error: 
> [2022.08.18-15.58.58:154][  0]LogExit: Executing StaticShutdownAfterError
> [2022.08.18-15.58.58:156][  0]LogWindows: FPlatformMisc::RequestExit(1)
> [2022.08.18-15.58.58:156][  0]LogWindows: FPlatformMisc::RequestExitWithStatus(1, 3)
> [2022.08.18-15.58.58:156][  0]LogCore: Engine exit requested (reason: Win RequestExit)
> [2022.08.18-15.58.58:164][  0]Log file closed, 08/18/22 17:58:58
>

**Comment by QuentinGarou** [comment]
After a few more tests it seems to appear randomly on startup no matter how I open it...

**Comment by dbaldassi** [comment]
This should now be fixed in [0d6b501](https://github.com/millicast/millicast-player-unreal-engine-plugin/commit/0d6b501d28f4179363b2e6f30ec1320a67ddb846)
